control-center/include/etc/confluent/docker/control-center.properties.template

```diff
@@ -52,18 +53,80 @@
+{% endfor -%}
+{% endmacro -%}
+
+{# ********************************************************************************************** #}
+{# SET_PROPERTIES_WITH_ENV_TO_PROPS_WITH_TWO_PREFIXES should be used for properties that have two #}
+{# fixed translations to the env variables. For example, for metrics properties, both #}
+{# CONTROL_CENTER_METRICS_ and CONFLUENT_METRICS_ deterministically always translate to #}
+{# confluent.metrics. However, the first env prefix takes precedence. Therefore, this function #}
+{# only sets a property that starts with the secondary env prefix if the property hasn't been set #}
+{# with the primary env prefix. We shouldn't find two copies of the same property being translated #}
+{# by both the primary and the secondary env prefix. Each translated property is NOT appended to #}
+{# excludes. #}
+{# Note this func could be deleted if env_to_props appends each translated property to excludes. #}
+{# ********************************************************************************************** #}
+{% macro SET_PROPERTIES_WITH_ENV_TO_PROPS_WITH_TWO_PREFIXES(primary_env_prefix, secondary_env_prefix, prop_prefix, excludes=[]) -%}
+{% set primary_props = env_to_props(primary_env_prefix, prop_prefix, exclude=excludes) -%}
+{% set secondary_props = env_to_props(secondary_env_prefix, prop_prefix, exclude=excludes) -%}
+{# SET PROPERTIES PREFIXED WITH primary_env_prefix BECAUSE THEY ALWAYS TAKE PRECEDENCE #}
+{% for name, value in primary_props.items() -%}
+{{name}}={{value}}
+{% endfor -%}
+{# SET PROPERTIES PREFIXED WITH secondary_env_prefix ONLY IF THE PROP HASN'T BEEN SET YET #}
+{% for name, value in secondary_props.items() -%}
+{% if name not in primary_props.keys() -%}
+{{name}}={{value}}
+{% endif -%}
+{% endfor -%}
+{% endmacro -%}
+
+{# ********************************************************************************************** #}
+{# SET_PROPERTIES_WITH_SKIP_PROP_CHECK should be used for properties that in general have a fixed #}
+{# translation, but have a few props prefixes that should be skipped. For example, properties #}
+{# that start with CONTROL_CENTER_ in general translates to confluent.controlcenter. However, #}
+{# CONTROL_CENTER_METRICS_* and CONTROL_CENTER_MONITORING_INTERCEPTOR_* also start with #}
+{# CONTROL_CENTER_ but they are special and should be skipped. Each translated property is NOT
+{# appended to excludes. #}
+{# Note this func could be deleted if env_to_props appends each translated property to excludes. #}
+{# ********************************************************************************************** #}
+{% macro SET_PROPERTIES_WITH_SKIP_PROP_CHECK(env_prefix, prop_prefix, excludes=[], skip_prop_prefix=[], skip_props=[]) -%}
+{% set props = env_to_props(env_prefix, prop_prefix, exclude=excludes) -%}
+{# CHECK IF NAME STARTS WITH A PROPERTY PREFIX THAT SHOULD BE SKIPPED #}
+{% for name, value in props.items() -%}
+{% for w in skip_prop_prefix -%}
+{% if w in name %}
```

Comment:
In our case, we expect if this is true then the name starts with `w` right? Alternatively, could we have used `startswith`?